utils/postgres.py

```diff
@@ -59,26 +67,26 @@ def copy_file(self, file: File, table: str, has_header: bool) -> list[dict]:
                 ),
                 file=open(os.path.join(file["source_path"], file["source_name"]), "r"),
             )
-            data = return_sql_results(cur)
+            data = self._return_sql_results(cur)
             self.conn.commit()
         return data
 
+    @staticmethod
+    def _return_sql_results(cur) -> list[dict]:
+        """Return data from a sql query
 
-def return_sql_results(cur) -> list[dict]:
-    """Return data from a sql query
-
-    Args:
-        cur (Cursor): cursor from postgres connection
+        Args:
+            cur (Cursor): cursor from postgres connection
 
-    Returns:
-        dict, bool: result of sql query or None
-    """
-    try:
-        data = cur.fetchall()
-    except psycopg2.ProgrammingError as e:
-        print(e)
-        data = None
-    if data:
-        columns = [desc[0] for desc in cur.description]
-        return [{k: v for k, v in zip(columns, d)} for d in data]
-    return []
+        Returns:
+                dict, None: result of sql query or None
```

Comment:
Actually now only returns a (potentially empty) list
```suggestion
                list: result of sql query, or an empty list if no result
```

Reply:
Good catch! We both fixed the issue `psycopg2.ProgrammingError` but my rebase was incomplete sorry! I like your solution to return an empty list instead of None. Really clean!